### ISSUE TITLE
Filenames containing combining characters cause misaligned display

### ISSUE THREAD

**Issue by craigbarnes** [ranger/ranger#1256]
#### Runtime Environment

- Operating system and version: Arch Linux
- Terminal emulator and version: Various; urxvt 9.22, xfce4-terminal 0.8.7.4, etc.
ranger version: ranger-master 1.9.1
Python version: 3.7.0 (default, Jul 15 2018, 10:44:58) [GCC 8.1.1 20180531]
Locale: en_GB.UTF-8

#### Current Behavior

Filenames containing combining or other non-spacing characters caused the file size column to become unaligned (shift more and more to the left, depending on the number of combining characters)

#### Expected Behavior

File size text aligned to the right edge of the column, as ASCII-only filenames are.

#### Possible Solutions

Use `wcwidth(3)` or similar to compute the number of terminal columns a filename occupies.

#### Steps to reproduce
1. `touch ฮั่ฮั่ฮั่ฮั่ฮั่ฮั่ฮั่ฮั่ฮั่ฮั่ฮั่ฮั่ฮั่`
2. `ranger`
3. Observe that the file size text is shifted too far left, due to miscalculated display width

**Comment by toonn** [comment]
Can confirm on the latest master. Thanks for the report!

**Comment by Fujiwara-Chika** [comment]
![image](https://user-images.githubusercontent.com/44722787/147486564-1a28ad02-57fd-442b-9969-b0d7034117d8.png)

Still not resolve?

**Comment by toonn** [comment]
No, that's why the issue is still open. The available fix isn't universal and I don't want to complicate matters while introducing more such subtle bugs.

**Comment by jquast** [comment]
I might be able to help with this, it looks like @qrvd and I have already discussed it lately, I have been reviewing projects for proper wcswidth/grapheme integration this week, this looks like a popular tool so let's make it CJK / emoji friendly :)
